docs/docs/backup-restore.md

```diff
@@ -0,0 +1,118 @@
+
+## Backup
+
+A backup operation reads the existing state from Marathon in zk and writes this data to a storage location.
+This data contains all entities under `zk://host:port/path/state/`. 
+It will not backup the state of the leader election: `zk://host:port/path/leader/`
+
+A backup is a single file in [tar](http://www.gnu.org/software/tar/) format, which contains all elements as file entries.
+
+### Options to create a backup 
+
+There are multiple ways to create a backup:
+
+- cmd line flag `--backup_location {URI}`
+Marathon can be started with the command line option: `--backup_location` to a configurable URI. 
+This option is especially useful and strongly recommended to set, since it will create a backup 
+of the Marathon internal state, whenever Marathon performs a migration. 
+A migration usually happens during an upgrade of Marathon.
+
+- API endpoint `DELETE /v2/leader?backup={URI}` (experimental)
+It is possible to create a backup at any point in time using the leader endpoint.
+Using this endpoint will force the current running leader to abdicate.
+The next elected leader will create a backup before this instance takes over the leadership.
+This method is marked experimental - the API endpoint may change in the future.
+
+- Executable packaged with Marathon
+The marathon native package ships with a start-script that can be used to create a backup.
+Please note: this command will connect directly to zk and backup the state from the defined location. 
+Use this method only, if there is no Marathon instance running!
+
+
+```
+$> marathon -main mesosphere.marathon.core.storage.backup.Backup --help
+$> marathon -main mesosphere.marathon.core.storage.backup.Backup --backup_location {backup_uri} --zk {zk_uri}
+```
+
+
+## Restore
+
+A backup that has been created, can be restored. Every restore operation will perform the following steps:
+- the current state in zk under the defined path will be removed
+- the state from the backup will be restored to the defined zk path 
+- the next Marathon leader will migrate this state to the current version
+
+### Options to restore from a backup 
+
+There are multiple ways to restore a backup:
+
+- API endpoint `DELETE /v2/leader?restore={URI}` (experimental)
+It is possible to instruct a running marathon instance to restore its state from a backup.
+Using this endpoint will force the current running leader to abdicate.
+The newly elected leader will perform a clean and restore operation, do a migration (if needed) and start up.
+This method is marked experimental - the API endpoint may change in the future.
+
+- Executable packaged with Marathon
+The marathon native package ships with a start-script that can be used to restore a backup.
+Please note: this command will connect directly to zk and restore the state from the defined location. 
+Use this method only, if there is no Marathon instance running!
```

Comment:
"The default Marathon package ships with a start script you can use to restore from a backup.

**Note:** This command connects directly to ZooKeepter and restores the state from the location you defined. Only use this method if there is no Marathon instance running."